modules/misc/xdg.nix

```
@@ -80,18 +80,21 @@ in {
   };
 
   config = mkMerge [
-    (mkIf cfg.enable {
-      xdg.cacheHome = mkDefault defaultCacheHome;
-      xdg.configHome = mkDefault defaultConfigHome;
-      xdg.dataHome = mkDefault defaultDataHome;
-      xdg.stateHome = mkDefault defaultStateHome;
-
-      home.sessionVariables = {
+    (let
+      variables = {
         XDG_CACHE_HOME = cfg.cacheHome;
         XDG_CONFIG_HOME = cfg.configHome;
         XDG_DATA_HOME = cfg.dataHome;
         XDG_STATE_HOME = cfg.stateHome;
       };
+    in mkIf cfg.enable {
+      xdg.cacheHome = mkDefault defaultCacheHome;
+      xdg.configHome = mkDefault defaultConfigHome;
+      xdg.dataHome = mkDefault defaultDataHome;
+      xdg.stateHome = mkDefault defaultStateHome;
+
+      home.sessionVariables = variables;
+      systemd.user.sessionVariables = mkIf pkgs.stdenv.isLinux variables;
```

```suggestion
      systemd.user.sessionVariables = mkIf pkgs.stdenv.hostPlatform.isLinux variables;
```

Isn't `stdenv.isLinux` an alias for `stdenv.hostPlatform.isLinux`?

IIRC there was a PR in HM that changed the former expression to the latter.

---

I copied what I saw in `modules/systemd.nix`. Grepping for `isLinux` it seems about 50/50. I can't find that PR, I'd be curious what the motivation is; is the alias discouraged?

---

Maybe you're thinking of https://github.com/nix-community/home-manager/pull/2528 ? Or https://github.com/nix-community/home-manager/pull/2419 ?

---

I fixed it for this file.